javascript/src/seedsource/containers/Upload.js

```diff
@@ -0,0 +1,210 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import shp from 'shpjs'
+import { updateConstraintValues } from '../../actions/constraints'
+import { setError } from '../../actions/error'
+
+class Upload extends React.Component {
+    constructor(props) {
+        super(props)
+
+        // Would this be useful for reports?
+        // this.state = {
+        //     filename: '',
+        //     uploadMessages: {},
+        //     shapeType: ''
+        // }
+
+        this.handleFileUpload = this.handleFileUpload.bind(this)
+        this.handleUploadFinish = this.handleUploadFinish.bind(this)
+    }
+
+
+    handleFileUpload(event) {
+        event.stopPropagation()
+        event.preventDefault()
+
+        document.getElementById("loadShapefileModal").style.display = "inherit"
```

Comment:
Interacting with the real, on-page DOM is a last resort. Is there no other way to do this? Note that there are some modal components that might be useful here.